tool/bindings_generator/generate_bindings.mjs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import * as fs from 'fs';
+import * as child_process from 'child_process';
+import { createRequire } from 'module';
+import * as idl from '@webref/idl';
+import * as assumedTypesModule from './assumedTypes.mjs';
+import * as translatorModule from './translator.mjs';
+import * as util from './util.mjs';
+
+const require = createRequire(import.meta.url);
+const Translator = translatorModule.Translator;
+const assumedTypesBlob = assumedTypesModule.assumedTypesBlob;
+
+function generateHeader() {
+  return (
+`// Copyright (c) ${util.year}, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// TODO(joshualitt): Import js types properly.
+import 'dart:typed_data';
+import 'package:js/js.dart';
+`);
+}
+
+async function translateIDLs() {
+  const files = await idl.listAll();
+  const translator = new Translator();
+  for (const [shortname, file] of Object.entries(files)) {
```

Comment:
Main suggestion (OK to do separately): we'd like to generate multiple libraries instead of one monolithic library. It helps with eng tooling (e.g. review tools cannot open files this big), but also with API discovery for users.

Reply:
one mini-lib for each file the exports would be GREAT!

Reply:
Let's do this in a follow on. Create #3.